src/Comparator/BackwardsCompatibility/ClassBased/MethodChanged.php

```diff
@@ -0,0 +1,58 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Roave\ApiCompare\Comparator\BackwardsCompatibility\ClassBased;
+
+use Roave\ApiCompare\Changes;
+use Roave\ApiCompare\Comparator\BackwardsCompatibility\MethodBased\MethodBased;
+use Roave\BetterReflection\Reflection\ReflectionClass;
+use Roave\BetterReflection\Reflection\ReflectionMethod;
+use function array_combine;
+use function array_intersect_key;
+use function array_keys;
+use function array_map;
+use function array_reduce;
+use function strtolower;
+
+final class MethodChanged implements ClassBased
+{
+    /** @var MethodBased */
+    private $checkMethod;
+
+    public function __construct(MethodBased $checkMethod)
```

Comment:
I'm liking this new nested structure! A little more in depth to start with, but it makes a lot of sense :+1:

Reply:
Incidentally; could `ClassBased` be a "child" of `InterfaceBased`, considering a class is a specialisation of an interface, perhaps?